appsec/tests/helper/client_test.cpp

```diff
@@ -1763,6 +1765,240 @@ TEST(ClientTest, RequestExecWithAttack)
+        EXPECT_STREQ(msg_res->actions[0].verdict.c_str(), "ok");
+
+        EXPECT_TRUE(std::regex_match(
+            msg_res->meta["_dd.appsec.fp.http.endpoint"].c_str(),
+            std::regex("\"http-get(-[A-Za-z0-9]*){3}\"")));
+
+        EXPECT_TRUE(std::regex_match(
+            msg_res->meta["_dd.appsec.fp.http.network"].c_str(),
+            std::regex("\"net-[0-9]*-[a-zA-Z0-9]*\"")));
+
+        EXPECT_TRUE(
+            std::regex_match(msg_res->meta["_dd.appsec.fp.http.header"].c_str(),
+                std::regex("\"hdr(-[0-9]*-[a-zA-Z0-9]*){2}\"")));
+
+        EXPECT_TRUE(
+            std::regex_match(msg_res->meta["_dd.appsec.fp.session"].c_str(),
+                std::regex("\"ssn(-[a-zA-Z0-9]*){4}\"")));
+    }
+}
+
+TEST(ClientTest, RequestShutdownWithFingerprint)
+{
+    auto smanager = std::make_shared<service_manager>();
+    auto broker = new mock::broker();
+
+    client c(smanager, std::unique_ptr<mock::broker>(broker));
+
+    set_extension_configuration_to(broker, c, EXTENSION_CONFIGURATION_ENABLED);
+    request_init(broker, c);
+
+    // Request Shutdown
+    {
+        network::request_shutdown::request msg;
+
+        msg.data = parameter::map();
+        msg.data.add("http.client_ip", parameter::string("192.168.1.1"sv));
+
+        // Endpoint Fingerprint inputs
+        auto query = parameter::map();
+        query.add("query", parameter::string("asdfds"sv));
+        msg.data.add("server.request.uri.raw", parameter::string("asdfds"sv));
+        msg.data.add("server.request.method", parameter::string("GET"sv));
+        msg.data.add("server.request.query", std::move(query));
+
+        // Network and Headers Fingerprint inputs
+        msg.data.add(
+            "server.request.headers.no_cookies_fp", parameter::string(""sv));
+
+        // Session Fingerprint inputs
+        msg.data.add("server.request.cookies", parameter::string("asdfds"sv));
+        msg.data.add("usr.session_id", parameter::string("asdfds"sv));
+        msg.data.add("usr.id", parameter::string("asdfds"sv));
+
+        network::request req(std::move(msg));
+
+        std::shared_ptr<network::base_response> res;
+        EXPECT_CALL(*broker, recv(_)).WillOnce(Return(req));
+        EXPECT_CALL(*broker,
+            send(
+                testing::An<const std::shared_ptr<network::base_response> &>()))
+            .WillOnce(DoAll(testing::SaveArg<0>(&res), Return(true)));
+
+        EXPECT_TRUE(c.run_request());
+        auto msg_res =
+            dynamic_cast<network::request_shutdown::response *>(res.get());
+        EXPECT_STREQ(msg_res->actions[0].verdict.c_str(), "block");
+
+        EXPECT_TRUE(std::regex_match(
+            msg_res->meta["_dd.appsec.fp.http.endpoint"].c_str(),
+            std::regex("\"http-get(-[A-Za-z0-9]*){3}\"")));
+
+        EXPECT_TRUE(std::regex_match(
+            msg_res->meta["_dd.appsec.fp.http.network"].c_str(),
+            std::regex("\"net-[0-9]*-[a-zA-Z0-9]*\"")));
```

Comment:
You can already do this in Gtest directly, with something along the lines of:

```cpp
EXPECT_THAT(value, MatchesRegex("http-get(-[A-Za-z0-9]*){3}"));
```

More info: https://google.github.io/googletest/reference/matchers.html